src/keri/vc/protocoling.py

```diff
@@ -405,3 +405,221 @@ def presentationExchangeExn(hab, reger, said):
     del ims[:exn.size]
 
     return exn, ims
+
+
+class IpexHandler(doing.Doer):
+    """ Processor of `exn` IPEX messages.
+
+    """
+
+    resource = "/ipex"
+    persist = True
+
+    def __init__(self, hby, rgy, notifier, **kwa):
+        """ Initialize instance
+
+        Parameters:
+            hab (Habitat): local identifier environment
+            wallet (Wallet) credential wallet that will hold the credentials
+            ims (Optional(bytearray)): inbound message stream to process
+            notifier (Notifier): outbound notifications
+            **kwa (dict): keyword arguments passed to DoDoer
+
+        """
+        self.hby = hby
+        self.rgy = rgy
+        self.notifier = notifier
+        self.msgs = decking.Deck()
+        self.cues = decking.Deck()
+
+        super(IssueHandler, self).__init__(**kwa)
+
+    def recur(self, tyme):
+        if self.msgs:
+            msg = self.msgs.popleft()
+            serder = msg["serder"]
+            attrs = serder.ked["a"]
+
+            if not attrs["r"].startWith(self.resource):
+                return False
+            data = dict(
+                r=f"/exn{attrs['r']}",
+                d=serder.said,
+                m=attrs["m"]
+            )
+
+            self.notifier.add(attrs=data)
+
+        return False
+    
+def ipexApplyExn(hab, message, schema, attrs):
+    """ Apply for an ACDC
+
+    Parameters:
+        hab(Hab): identifier environment for issuer of credential
+        message(str): Human readable message regarding the credential application
+        schema (any): schema or its SAID
+        attrs (any): attribute field label list
+
+    Returns:
+        Serder: credential issuance exn peer to peer message
+        bytes: attachments for exn message
+
+    """
+    data = dict(
+        m=message,
+        s=schema,
+        a=attrs
+    )
+
+    exn, end = exchanging.exchange(route="/ipex/apply", payload=data, sender=hab.pre)
+    ims = hab.endorse(serder=exn, last=False, pipelined=False)
+    del ims[:exn.size]
+    ims.extend(end)
+
+    return exn, ims
+
+def ipexOfferExn(hab, message, offer):
```

Comment:
I think the offer should include the ACDC and it should be signed and embedded.

Reply:
the offer is the metadata ACDC. I'll add as signed and embedded. Should it be called `offer` or `acdc`? 
We can also add the said in the payload.

Reply:
Either is fine but I like acdc better.

Reply:
changed to acdc